usb/backend/libusb1.py

```diff
@@ -651,6 +700,107 @@ def __init__(self, dev):
         self.devid = dev.devid
         _check(_lib.libusb_open(self.devid, byref(self.handle)))
 
+
+class _AsyncTransfer():
+    """ Generic API for asynchronous USB transfers"""
+    def __init__(self, dev_handle, ep, buf, iso_packet_count, timeout):
+        self._transfer = _lib.libusb_alloc_transfer(iso_packet_count)
+        self._ctx = None
+        self._buf = buf
+
+    def __del__(self):
+        if not self._fut.done():
+            _lib.libusb_cancel_transfer(self._transfer)
+
+        _lib.libusb_free_transfer(self._transfer)
+
+    def submit(self, ctx = None):
+        """ Submit the USB transfer, but don't wait for completion
+
+        TODO: Is it better to create the Future in __init__ and reuse it?
+        TODO 2: Exception when libusb_submit_transfer() fails()
+        """
+        self._fut = asyncio.Future()
+        self._ctx = ctx
+        _check(_lib.libusb_submit_transfer(self._transfer))
+
+        return self
+
+    def wait_sync(self):
+        """ Wait synchronously on transfer shich has already been submitted.
+
+        TODO: Is this function needed? If the purpose is to make an async API,
+        then what is the purpose of providing a synchronous function?
+        """
+        asyncio.run(self._usb_event_loop())
+        return self.fut.result()
+
+    async def _usb_event_loop(self):
+        """ Helper function to process USB events and complete futures
+
+        TODO: This processes all events in the current libusb context, self._ctx
+        It seems benign, since transfers that complete earlier will have been
+        marked as donem and not run libusb_handle_events(). However, is this a
+        correct way to handle events?
+        TODO 2: Is libusb_handle_events() really non-blocking?
```

Comment:
Nope 😕 I tried to test on my receipt printer:
```python
from escpos.printer import Usb

p = Usb(0x2730, 0x2002, None, 0, 0x81, 0x02)
p.set('left')
p.text(f"{'ПУТИН ХУЙЛО ' * 300}")
p.qr("All the sings she said", size=16, model=2)
p.cut(feed=False)
````

Changed the function to show execution time:
```python
        import time
        while not self._fut.done():
            st = time.time()
            _check(_lib.libusb_handle_events(self._ctx))
            print(round(time.time()-st, 5))
            await asyncio.sleep(0.0001)
```

```
6e-05
4e-05
3e-05
3e-05
3e-05
3e-05
3e-05
5e-05
3e-05
1e-05
# start of printing qr
0.01008
# end of printing qr
1e-05
# start of printing "путин хуйло"
6.02684
# end of printing "путин хуйло"
4e-05
3e-05
3e-05

```